### PR TITLE
Offline Mode: Add a way to cancel uploads directly from Media Uploads screen

### PR DESCRIPTION
**Pull request by kean**
Add a way to cancel uploads directly from the "Media Uploads" screen.

Note: this doesn't cover drafts sync yet, and I'm working on integrating this screen in more context, but wanted to keep the PRs small.

https://github.com/wordpress-mobile/WordPress-iOS/assets/1567433/9dae6e7c-6605-452d-a903-5b561ee3c6f0

https://github.com/wordpress-mobile/WordPress-iOS/assets/1567433/550a3bf6-c48a-4cb1-8e37-2b4ff7dcfe8a

## To test:

- Use the steps from the [following issue](https://github.com/wordpress-mobile/WordPress-iOS/pull/23091#issuecomment-2078454749) to create a post with a dangling media upload
- Tap "Publish"
- ✅ Verify that the sheet shows a "Media failed to upload message" (it is a defect, but for the purposes of this PR, it's an expected behavior)
- Open "Media Uploads"
- Tap "More" and Tap "Cancel Upload" on the danling upload
- Tap "Back"
- ✅ Verify that the "Publish" button is now available and you can publish the post

## Regression Notes
1. Potential unintended areas of impact

2. What I did to test those areas of impact (or what existing automated tests I relied on)

3. What automated tests I added (or what prevented me from doing so)

PR submission checklist:

- [ ] I have completed the Regression Notes.
- [ ] I have considered adding unit tests for my changes.
- [ ] I have considered adding accessibility improvements for my changes.
- [ ] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

Testing checklist:
- [ ] WordPress.com sites and self-hosted Jetpack sites.
- [ ] Portrait and landscape orientations.
- [ ] Light and dark modes.
- [ ] Fonts: Larger, smaller and bold text.
- [ ] High contrast.
- [ ] VoiceOver.
- [ ] Languages with large words or with letters/accents not frequently used in English.
- [ ] Right-to-left languages. (Even if translation isn’t complete, formatting should still respect the right-to-left layout)
- [ ] iPhone and iPad. 
- [ ] Multi-tasking: Split view and Slide over. (iPad)

### CODE REVIEW

**Comment by dvdchr** [comment]
Hi @kean 👋 , I'm bumping this PR's milestone to `24.9` since I'm starting code freeze. Feel free to re-target this to the release branch if this is a blocker or intended for `24.8`.

File: WordPress/Classes/Services/MediaCoordinator.swift [hunk 1]
```diff
@@ -588,6 +591,8 @@ class MediaCoordinator: NSObject {
         case ended
         case failed(error: NSError)
         case progress(value: Double)
+        /// The upload was cancelled by the user.
+        case cancelled
```
Comment (momo-ozawa) [inline WordPress/Classes/Services/MediaCoordinator.swift:595]: [nit] Let's use the US spelling for canceled
```suggestion
        case canceled
```

File: WordPress/Classes/Services/MediaCoordinator.swift [hunk 2]
```diff
@@ -603,6 +608,8 @@ class MediaCoordinator: NSObject {
                 return "Failed: \(error)"
             case .progress(let value):
                 return "Progress: \(value)"
+            case .cancelled:
+                return "Cancelled"
```
Comment (momo-ozawa) [inline WordPress/Classes/Services/MediaCoordinator.swift:612]: [nit] Let's use the US spelling for canceled
```suggestion
        case canceled
```
```suggestion
            case .canceled:
                return "Canceled"
```

**Review by momo-ozawa — APPROVED** [review]
LGTM w a few nits

Wdyt of reordering / changing the copy for this sheet to
* `Retry upload`
* `Stop upload`
to keep things consistent with the other changes you made?

File: WordPress/Classes/Services/MediaCoordinator.swift [hunk 3]
```diff
@@ -282,6 +282,9 @@ class MediaCoordinator: NSObject {
     func cancelUploadAndDeleteMedia(_ media: Media) {
         cancelUpload(of: media)
         delete(media: [media])
```
Comment (momo-ozawa) [inline WordPress/Classes/Services/MediaCoordinator.swift:284]: I just noticed that ` delete(media: [media])` calls `cancelUpload(of: media)`

https://github.com/wordpress-mobile/WordPress-iOS/blob/cfee33ebc58eb0ccef2d35c93d9e420948caae83/WordPress/Classes/Services/MediaCoordinator.swift#L322-L323

Comment (kean) [reply WordPress/Classes/Services/MediaCoordinator.swift:284]: Yeah, you are right. It is a no-op when called second time, so it should be OK. I left a note to address it separately.

File: WordPress/Classes/Services/MediaCoordinator.swift (hunk 1 above)
Comment (kean) [reply WordPress/Classes/Services/MediaCoordinator.swift:595]: I think it's a toss-up. Apple uses the double "LL" version throughout their APIs too. Example: https://developer.apple.com/documentation/swift/task/iscancelled-swift.type.property

**Comment by dangermattic** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">This PR is assigned to the milestone <a href="https://github.com/wordpress-mobile/WordPress-iOS/milestone/286">24.8 ❄️</a>. The due date for this milestone has already passed.<br />
Please assign it to a milestone with a later deadline or check whether the release for this milestone has already been finished.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

**Pull request merged**